### PR TITLE
Fixing structured dataset for empty bq tables

### PR DESCRIPTION
**Pull request by dylanwilder**
# TL;DR
Structured dataset bq read fails on empty table because `pandas.concat()` fails if the passed array is empty.  Fix it to create an empty dataframe with the required schema

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Review by eapolinario — COMMENTED** [review]
Can you add a unit-test for the empty case [here](https://github.com/flyteorg/flytekit/blob/master/tests/flytekit/unit/types/structured_dataset/test_bigquery.py)?

**Review by kumare3 — APPROVED** [review]

File: flytekit/types/structured/bigquery.py [hunk 1]
```diff
@@ -51,7 +51,14 @@ def _read_from_bq(
     frames = []
     for message in reader.rows().pages:
         frames.append(message.to_dataframe())
-    return pd.concat(frames)
+    if len(frames) > 0:
+        df = pd.concat(frames)
```
Comment (pingsutw) [inline flytekit/types/structured/bigquery.py:55]: If the table is empty, the length of `read_session.streams` is 0. so I think we should update the code to below.
Let me know if it doesn't work for you
```suggestion
    frames = []
    if len(read_session.streams) > 0:
        stream = read_session.streams[0]
        reader = client.read_rows(stream.name)
        for message in reader.rows().pages:
            frames.append(message.to_dataframe())

    if len(frames) > 0:
        df = pd.concat(frames)
```

Comment (wild-endeavor) [reply flytekit/types/structured/bigquery.py:55]: @dylanwilder could you take a look at this snippet please? Does this work for you?

Comment (gitgraghu) [reply flytekit/types/structured/bigquery.py:55]: @wild-endeavor I tested this snippet. Seems like there will be a stream even for empty tables. So this does not work.
